crates/bevy_asset/src/batch.rs

```diff
@@ -0,0 +1,31 @@
+use alloc::vec::Vec;
+
+use crate::{Asset, AssetPath, UntypedHandle};
+use bevy_reflect::TypePath;
+
+pub struct LoadBatchRequest {
+    pub requests: Vec<AssetPath<'static>>,
+}
+
+impl LoadBatchRequest {
+    pub fn new<T>(requests: Vec<T>) -> Self
+    where
+        T: Into<AssetPath<'static>>,
+    {
+        Self {
+            requests: requests.into_iter().map(Into::into).collect(),
+        }
+    }
```

Comment:
Nit, but nicer (allows more than just `vec![foo, bar]`):
```suggestion
    pub fn new<T>(requests: impl IntoIterator<Item = T>) -> Self
    where
        T: Into<AssetPath<'static>>,
    {
        Self {
            requests: requests.into_iter().map(Into::into).collect(),
        }
    }
```